railties/lib/rails/application/bootstrap.rb

```diff
@@ -39,7 +39,11 @@ module Bootstrap
           f.binmode
           f.sync = config.autoflush_log # if true make sure every write flushes
 
-          logger = ActiveSupport::Logger.new f
+          logger_args = [f]
+          logger_args << config.logger_shift_age
+          logger_args << config.logger_shift_size if config.logger_shift_age.to_i > 0
+
+          logger = ActiveSupport::Logger.new(*logger_args.compact)
```

Comment:
Why do we need `compact` here? If the arguments are ordered and one is nil, then we'd submit different arguments to `new`.